### ISSUE TITLE
why the get_gradient_norm return none?

### ISSUE THREAD

**Issue by thbupt** [a-r-r-o-w/cogvideox-factory#405]
### System Info / 系統信息

transformers              4.49.0
torch                     2.6.0
deepspeed                 0.16.4
accelerate                1.5.2
diffusers                 0.33.1

### Information / 问题信息

- [ ] The official example scripts / 官方的示例脚本
- [ ] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

def get_gradient_norm(parameters):
    norm = 0
    for param in parameters:
        if param.grad is None:
            continue
        local_norm = param.grad.detach().data.norm(2)
        norm += local_norm.item() ** 2
    norm = norm**0.5
    return norm

if accelerator.sync_gradients:
            gradient_norm_before_clip = get_gradient_norm(transformer.parameters())
            accelerator.clip_grad_norm_(transformer.parameters(), args.max_grad_norm)
            gradient_norm_after_clip = get_gradient_norm(transformer.parameters())

### Expected behavior / 期待表现

When I use the original training code for cogvideox, I found that the gradients before and after clip are all none.
So I guess the clip_grad_norm_ doesn't work. But why?
